src/event_listener/shared.rs

```diff
@@ -685,236 +674,311 @@ macro_rules! report_unknown {
+                ParsedEventType::FullscreenStateChanged,
+                r"fullscreen>>(?P<state>0|1)"
+            ),
+            (
+                ParsedEventType::MonitorRemoved,
+                r"monitorremoved>>(?P<monitor>.*)"
+            ),
+            (
+                ParsedEventType::MonitorAdded,
+                r"monitoradded>>(?P<monitor>.*)"
+            ),
+            (
+                ParsedEventType::WindowOpened,
+                r"openwindow>>(?P<address>.*),(?P<workspace>.*),(?P<class>.*),(?P<title>.*)"
+            ),
+            (
+                ParsedEventType::WindowClosed,
+                r"closewindow>>(?P<address>.*)"
+            ),
+            (
+                ParsedEventType::WindowMoved,
+                r"movewindow>>(?P<address>.*),(?P<workspace>.*)"
+            ),
+            (
+                ParsedEventType::LayoutChanged,
+                r"activelayout>>(?P<keyboard>.*)(?P<layout>.*)"
+            ),
+            (ParsedEventType::SubMapChanged, r"submap>>(?P<submap>.*)"),
+            (
+                ParsedEventType::LayerOpened,
+                r"openlayer>>(?P<namespace>.*)"
+            ),
+            (
+                ParsedEventType::LayerClosed,
+                r"closelayer>>(?P<namespace>.*)"
+            ),
+            (
+                ParsedEventType::FloatStateChanged,
+                r"changefloatingmode>>(?P<address>.*),(?P<floatstate>[0-1])"
+            ),
+            (
+                ParsedEventType::Minimize,
+                r"minimize>>(?P<address>.*),(?P<state>[0-1])"
+            ),
+            (
+                ParsedEventType::Screencast,
+                r"screencast>>(?P<state>[0-1]),(?P<owner>[0-1])"
+            ),
+            (
+                ParsedEventType::UrgentStateChanged,
+                r"urgent>>(?P<address>.*)"
+            ),
+            (
+                ParsedEventType::WindowTitleChanged,
+                r"windowtitle>>(?P<address>.*)"
+            ),
+            (ParsedEventType::Unknown, r"(?P<Event>[^>]*)"),
+        ]
+        .into_iter()
+        .map(|(e, r)| (e, check_for_regex_error(Regex::new(r))))
+        .collect();
     }
 
     let event_iter = event.trim().split('\n');
 
     let mut events: Vec<Event> = vec![];
 
     for item in event_iter {
-        let matches = EVENT_SET.matches(item);
-        let matches_event: Vec<_> = matches.into_iter().collect();
-        let captures = if !EVENT_REGEXES.is_empty() && !matches_event.is_empty() {
-            match EVENT_REGEXES[matches_event[0]].captures(item) {
-                Some(captures) => captures,
-                None => panic!("Regex has no captures"),
+        let matched: Vec<_> = EVENT_SET
+            .iter()
+            .filter(|(_, r)| r.is_match(item))
+            .map(|(pet, r)| {
+                (
+                    pet,
+                    r.captures(item)
+                        .unwrap_or_else(|| panic!("Regex has no captures")),
```

Comment:
Could this become recoverable? Or if not more clear since users will be the ones most likely to see this, and might not have any context

Reply:
This is unreachable code in theory, but in case, this prevent the developper of Hyprland-rs breaking the regex for unknown that there is a problem.